libs/libarchfpga/src/physical_types.h

```diff
@@ -1750,11 +1756,23 @@ struct t_arch {
     int num_switches;
     t_direct_inf* Directs = nullptr;
     int num_directs = 0;
+
     t_model* models = nullptr;
     t_model* model_library = nullptr;
+
     t_power_arch* power = nullptr;
     t_clock_arch* clocks = nullptr;
 
+    // Constants
+    std::string gnd_cell;
+    std::string vcc_cell;
+
+    std::string gnd_net = "$__gnd_net";
```

Comment:
So, in general, the name of the GND and VCC nets do come from the interchange device description, but they might be absent from the schema, as they are optional. For instance, the Xilinx device have the constant nets specified as `GLOBAL_LOGIC_0` and `GLOBAL_LOGIC_1`.

Given that these net names are variable from arch to arch, they need to be parased from the device file, and this would need to happen in the `libarchfpga` library.

Reply:
Make sense. Would you mind giving more insights about what client functions require these data? 
Currently I see it is used by netlist readers. I am sure it will be used by netlist writers.
When adding the information to the architecture database, we expect they may be used by packer, placer, router as well.

Reply:
Indeed, for now I only expect the constant nets to be used in the front-end and back-end only, but it is not excluded that they might come in handy in core algorithms, especially in the packer.

The main reason it is added to the architecture database is because of accessibility of the information stored in the [interchange device database](https://github.com/chipsalliance/fpga-interchange-schema/blob/53e3feda4628e39e939ce8d2e4add72025809d42/interchange/DeviceResources.capnp#L436-L540) and needs to be read in the netlist reader.

Reply:
I see. Maybe it is better to add additional comments to the source file, so that developers know what these variables are.
It is important to clarify 
- if the ``vcc_cell`` and ``gnd_cell`` physically exist in an FPGA fabric/architecture or that they are just virtual. I understand that in commerical FPGAs, there are dedicated circuitry to generate these constant signals.
- if the ``vcc_net`` and ``gnd_net`` are the required net names for input netlists (.blif,  .eblif etc.) to represent ``vcc`` and ``gnd`` signal. And just briefly explain why they should be in the architecture database and their relationship with the ``vcc_cell`` and ``gnd_cell``

Reply:
All right, I have added some extra comments around that, hopefully those help clarify a bit more on this.